--- a/vendor/filament/filament/src/Panel/Concerns/HasComponents.php
+++ b/vendor/filament/filament/src/Panel/Concerns/HasComponents.php
@@ -20,7 +20,6 @@ use Illuminate\Database\Eloquent\Model;
 use Illuminate\Filesystem\Filesystem;
 use Livewire\Component;
 use Livewire\Livewire;
-use Livewire\Mechanisms\ComponentRegistry;
 use ReflectionClass;
 
 trait HasComponents
@@ -591,7 +590,7 @@ trait HasComponents
 
     protected function queueLivewireComponentForRegistration(string $component): void
     {
-        $componentName = app(ComponentRegistry::class)->getName($component);
+        $componentName = app('livewire.factory')->resolveComponentName($component);
 
         $this->livewireComponents[$componentName] = $component;
     }
